### DISCUSSION TITLE
Run from systemd service - problem

### DISCUSSION THREAD

**Discussion by CwnAnnwn** [Geektoolkit/Dynaframe3#104] [General]
I've been trying to configure a systemd service for this, but cannot get the app to launch successfully. Any advice?

dynaframe.service
[Unit]
   Description = Dynaframe service
   After = graphical.target
   Requires = network.target

[Service]
   Type = idle
   User = pi
   WorkingDirectory = /home/pi/Dynaframe
   ExecStart = /home/pi/Dynaframe/Dynaframe
   Restart = always
   RestartSec = 10

[Install]
   WantedBy = graphical.target

**Comment by CwnAnnwn** [comment]
### Solved
**dynaframe.service**
`[Unit]
   Description = Dynaframe service
   After = graphical.target
   Requires = network.target

[Service]
   Type = idle
   User = pi
   Environment=DISPLAY=:0
   WorkingDirectory = /home/pi/Dynaframe
   ExecStart = /home/pi/Dynaframe/Dynaframe
   Restart = always
   RestartSec = 10

[Install]
   WantedBy = graphical.target`

**Comment by Geektoolkit** [comment]
Thankyou for this...question..by default it should be using autostart from the script to setup. Are you installing it manually? And is there a beneift of a systemd vs autostart? 
Ironically my linux knowlege is really ...elementary.  I'm working on learning it more, but I'm more of a windows dev that needed this to run on a pi.  So please forgive my probably basic questions as I try to understand this. I'd tried to set this up and wasn't successful at all just out of inexperience, so I'm glad to see you got it working. If nothing else I can add a wiki entry to document this for those that want to do setup from scratch for thier own OS/uses

**Reply by CwnAnnwn** [reply]
I removed that .desktop from ~/.config/autostart once I got this working. 

Systemd gives you more control over the process. It starts automatically, and restarts automatically if the program gets closed. You can start, stop, restart, or check the status of the service on demand. Even over SSH! Very handy!

E g. Sudo systemctl restart dynaframe

I too come from a Windows perspective: I work IT. As such I don't want Windows anywhere near what I do for fun.

Thank you for this utility, it was great to find it when I was looking for this functionality.

**Reply by CwnAnnwn** [reply]
I tweaked the dynaframe.service to use the run.sh script instead of running dynaframe directly. Works just fine!

[Service]
Type = idle
User = pi
Environment=DISPLAY=:0
WorkingDirectory = /home/pi/Dynaframe
ExecStart = /home/pi/Dynaframe/run.sh
Restart = always
RestartSec = 10
